components/core/src/clp/GlobalMetadataDBConfig.cpp

```diff
@@ -1,110 +1,99 @@
 #include "GlobalMetadataDBConfig.hpp"
 
+#include <cstdlib>
+#include <stdexcept>
+
 #include <fmt/core.h>
-#include <yaml-cpp/yaml.h>
 
-using std::exception;
 using std::invalid_argument;
 using std::string;
 
-static exception get_yaml_missing_key_exception(string const& key_name) {
-    throw invalid_argument(fmt::format("Missing key '{}'", key_name));
-}
+namespace clp {
+// Constants
+constexpr int cMinPort{1};
+constexpr int cMaxPort{65'535};
+
+auto operator>>(std::istream& in, GlobalMetadataDBConfig::MetadataDBType& metadata_db_type)
+        -> std::istream& {
+    string db_type_string;
+    in >> db_type_string;
+
+    if ("sqlite" == db_type_string) {
+        metadata_db_type = GlobalMetadataDBConfig::MetadataDBType::SQLite;
+    } else if ("mysql" == db_type_string) {
+        metadata_db_type = GlobalMetadataDBConfig::MetadataDBType::MySQL;
+    } else {
+        throw invalid_argument("Unknown database type: " + db_type_string);
```

Comment:
let's use fmt::format instead

```suggestion
        throw invalid_argument(fmt::format("Unknown database type: {}",db_type_string));
```